outline.md

```diff
@@ -0,0 +1,292 @@
+    - converting-a-text-mate-theme.md
+    - theme-variables.md
+    - Shadow DOM
+    - https://atom.io/docs/latest/upgrading/upgrading-your-ui-theme
+
+## Chapter 4: Behind Atom
+
+- advanced dev tools
+  - keybinding resolver
+  - timecop
+- Maintaining Packages
+  - how to publish
+    - Atom.io package system (adding, searching, stars)
+  - changelog
+  - issue tracker
+- Best practices (patterns and anti-patterns)
+  - aggregating subscriptions
+  - view best practices
+  - don't make a slow package
+  - CI for packages (https://blog.atom.io/2014/04/25/ci-for-your-packages.html)
+
+### Include in Chapter 3 or 4
+
+- Make sure we include (in Ch 3 or Ch4 or both)
+  - scopes (https://atom.io/docs/v0.174.0/advanced/scopes-and-scope-descriptors)
+  - views / models
+  - settings (how to enable the settings button on the package installer)
+  - notifications package
+  - config api
+    - https://blog.atom.io/2014/10/02/config-api-has-schema.html
+    - https://blog.atom.io/2014/10/31/language-scoped-config.html
+    - https://atom.io/docs/latest/advanced/scopes-and-scope-descriptors
+  - subscriptions
+    - https://blog.atom.io/2014/09/16/new-event-subscription-api.html
+    - subscribing to events
+    - subscribing to commands
+  - creating dom elements
+    - no longer spacepen
+  - runners
+  - configure my package
+  - add application menu/context menu
+  - get all the editors
+  - understand when the active editor changes
+  - background tasks
+  - messages to the user
+  - dialogs
+  - advanced keymaps (in Guides)
+  - serialization
+  - converting textmate bundles
+  - decoration (https://blog.atom.io/2014/07/24/decorations.html)
+  - modal panel
+  - modal decorations
+  - notifications
+
+## Chapter 5: Contributing to Atom
+
+- To packages
+  - Cloning and apm linking
+  - apm install/update
+  - Native modules
+  - Running specs
+  - Link to sending a pull request
+- To core
+  - Opening issues
+  - Roadmaps
+    - https://blog.atom.io/2014/10/23/public-roadmap.html
+  - Where is the code?
+  - Forum
+  - Building Atom
+  - Technologies it uses
+    - Node
+      - between here and npm, probably list the packages we make suse of and what they do in terms of the project. This is going to be where we need support from the Core and backend team.
+    - Atom-shell
+    - apm
+    - npm
+      - Etch with note we're transitioning to solidjs
+      - Electron
+  - Coding conventions
+  - Commit conventions
+
+## ~Chapter 6: Electron~
+
+This section is probably going to be relegated to not being implemented. We have no control over electron and the best we can do is point people to their website elsewhere.
+
+- ~developing a new project in electron~
+- ~https://github.com/atom/electron~
+
+## Chapter 7 / Appendix
+
+- Reference Manual
+- Cookbook snippets <- this should be moved and put elsewhere.
+  - As in this Is probably more tutorial related if we want to make a section based on tutorials and move them all there.
```

Comment:
That might fit in parantheses (the "As in this" one I'm relating to).

Reply:
I almost did, but this was another one of my notes on this for anyone looking at this PR. A lot of my time was spent just getting this to a point where I could put it up and ask for suggestions to reorganize it.